.github/workflows/nodejs.yml

```diff
@@ -1,31 +1,49 @@
-name: 'tests'
-
-on:
-  push:
-    branches: [master]
-  pull_request:
-    branches: [master]
+on: [push, pull_request]
 
 jobs:
-  tests:
-    runs-on: 'ubuntu-latest'
-
+  publish:
+    needs: ci
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      - uses: actions/setup-node@v3
+        with:
+          node-version: latest
+      - run: npm install
+      - run: npx semantic-release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+  ci:
+    needs: ci-matrix
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 0
+  ci-matrix:
+    needs: get-supported-node-versions
+    runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [12.x, 14.x, 16.x]
-
+        node-version: ${{ fromJson(needs.get-supported-node-versions.outputs.versions) }}
     steps:
-      - uses: 'actions/checkout@v3.0.2'
-
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: 'actions/setup-node@v3.5.0'
+        uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
-
-      - name: 'Install Dependencies'
-        run: 'npm install'
-
-      - run: 'npm run lint:commit -- --to "${{ github.sha }}"'
-      - run: 'npm run lint:standard'
-      - run: 'npm run lint:ts-standard'
-      - run: 'npm run test'
+      - run: npm install
+      - run: npm run lint:commit -- --from "7b4219a67b4fe94b1a1487b7d545d0cc7742f770" --to "${{ github.sha }}"
```

Comment:
Why adding this `from` flag with a hard-coded commit hash?

Reply:
This is because some old commits in the commit history don't follow the format and cause commitlint to error, as you can see below.
```
$ npm run lint:commit -- --to "79b63eb"

> ts-standard@12.0.3 lint:commit
> commitlint --to 79b63eb

⧗   input: feat(package): Allow multiple projects to be used (#163)
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]

✖   found 1 problems, 0 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint

⧗   input: fix(badges): Fix readme file badges
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]

✖   found 1 problems, 0 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint

⧗   input: fix(typescript): Try to make typescript a peerDependency
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]

✖   found 1 problems, 0 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint

⧗   input: Initial commit
✖   subject may not be empty [subject-empty]
✖   type may not be empty [type-empty]

✖   found 2 problems, 0 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint
```
I've hardcoded it from the latest commit on standard/ts-standard as that's the last commit before my commits, and this is when I've setup a precommit hook for commitlint.